tools/clang/lib/Sema/SemaOverload.cpp

```diff
@@ -5838,8 +5838,21 @@ Sema::AddOverloadCandidate(FunctionDecl *Function,
       // parameter of F.
       // HLSL Change Starts
       if (getLangOpts().HLSL) {
+        ParmVarDecl *Param = Function->getParamDecl(ArgIdx);
+        // extract allowable type info from param
+        unsigned legalTypes = Param->getLegalTypes();
```

Comment:
This is the place I need this information and I'm not sure if this is the right way to get it here.